src/visitors.rs

```diff
@@ -235,6 +248,33 @@ impl<'a, 'b: 'a, 'tcx: 'b> MirVisitor<'a, 'b, 'tcx> {
         self.summary_cache.set_summary_for(self.def_id, summary);
     }
 
+    /// Use the visitor to compute the state corresponding to promoted constants.
+    fn promote_constants(&mut self, mut state_with_parameters: Environment) -> Environment {
+        let saved_mir = self.mir;
+        let result_root = Path::LocalVariable { ordinal: 0 };
+        let mut ordinal = 0;
+        for constant_mir in self.mir.promoted.iter() {
+            self.mir = constant_mir;
+            self.visit_body();
+            let promoted_root = Path::PromotedConstant { ordinal };
+            let value = self.lookup_path_and_refine_result(result_root.clone());
+            state_with_parameters.update_value_at(promoted_root.clone(), value);
+            for (path, value) in self
+                .exit_environment
+                .value_map
+                .iter()
+                .filter(|(p, _)| Self::path_is_rooted_by(*p, &result_root))
+            {
+                let promoted_path = Self::replace_root(&path, &result_root, promoted_root.clone());
+                state_with_parameters.update_value_at(promoted_path, value.clone());
+            }
+            self.reset_visitor_state();
+            ordinal += 1;
```

Comment:
Maybe I just can't see it, but you're not doing anything with ordinal? If you were in the loop you could just enumerate(). Actually you could do that anyway, but set the value of ordinal to the value enumerate gives you, so you could use it after the loop.

Reply:
Will use enumerate.